diff -ruN linux-2.6.13/drivers/usb/input/Kconfig linux-2.6.13-synusb/drivers/usb/input/Kconfig
--- linux-2.6.13/drivers/usb/input/Kconfig	2005-12-03 01:50:05.351563984 +0100
+++ linux-2.6.13-synusb/drivers/usb/input/Kconfig	2005-12-03 01:50:44.713580048 +0100
@@ -272,3 +272,29 @@
 
 	  To compile this driver as a module, choose M here: the module will
 	  be called keyspan_remote.
+
+config USB_SYNAPTICS
+	tristate "Synaptics USB Touchpad support"
+	depends on USB && INPUT
+	---help---
+	  Say Y here if you want to use a Synaptics USB Touchpad.
+	  Synaptics USB touchpads emulate an USB mouse by default, so you
+	  can also use the usbhid driver. But you will be able to use
+	  Synaptics TouchPad driver for XOrg/XFree86 in combination with
+	  this kernel module, which will improve mouse movement, enable
+	  vertical/horizontal scrolling, etc. More Information can be
+	  found at:
+	  <http://www.uni-jena.de/~p1stja/linux/synaptics-usb.html>
+
+	  To compile this driver as a module, choose M here: the
+	  module will be called synaptics-usb.
+
+config USB_CPADDEV
+	bool "Synaptics cPad display support"
+	depends on USB_SYNAPTICS
+	help
+	  Say Y here if you want to use the display of Synaptics cPad.
+	  The display can be acessed through a character device
+	  /dev/usb/cpad?. Programs that use the character device can
+	  be found at:
+	  <http://www.uni-jena.de/~p1stja/linux/synaptics-usb.html>
diff -ruN linux-2.6.13/drivers/usb/input/Makefile linux-2.6.13-synusb/drivers/usb/input/Makefile
--- linux-2.6.13/drivers/usb/input/Makefile	2005-12-03 01:50:05.349564288 +0100
+++ linux-2.6.13-synusb/drivers/usb/input/Makefile	2005-12-03 01:50:05.357563072 +0100
@@ -40,3 +40,4 @@
 obj-$(CONFIG_USB_WACOM)		+= wacom.o
 obj-$(CONFIG_USB_ACECAD)	+= acecad.o
 obj-$(CONFIG_USB_XPAD)		+= xpad.o
+obj-$(CONFIG_USB_SYNAPTICS)	+= synaptics-usb.o
