diff -ruN linux-2.6.18-synusb/drivers/usb/input/synaptics-usb.c linux-2.6.18-synusb-cleanup/drivers/usb/input/synaptics-usb.c
--- linux-2.6.18-synusb/drivers/usb/input/synaptics-usb.c	2006-10-08 14:14:27.000000000 +0200
+++ linux-2.6.18-synusb-cleanup/drivers/usb/input/synaptics-usb.c	2006-10-08 14:16:10.000000000 +0200
@@ -22,11 +22,6 @@
  */
 
 
-#include "synusb-kcompat.h"
-
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,15)
-#include <linux/config.h>
-#endif
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
@@ -34,14 +29,8 @@
 #include <linux/module.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)
 #include <linux/usb/input.h>
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,13)
-#include <linux/usb_input.h>
-#endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,5)
 #include <linux/kref.h>
-#endif
 #include <linux/input.h>
 #include <linux/moduleparam.h>
 #include <linux/cpad.h>
@@ -266,11 +255,7 @@
 	idev->phys = synusb->iphys;
 	idev->name = synusb_input_name;
 	usb_to_input_id(udev, &idev->id);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,15)
 	idev->cdev.dev = &synusb->interface->dev;
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,2)
-	idev->dev = &synusb->interface->dev;
-#endif
 
 	input_register_device(idev);
 
@@ -717,9 +702,6 @@
 static struct usb_class_driver cpad_class = {
 	.name =		"usb/cpad%d",
 	.fops =		&cpad_fops,
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,15)
-	.mode =		S_IFCHR | S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP,
-#endif
 	.minor_base =	USB_CPAD_MINOR_BASE,
 };
 
@@ -957,9 +939,6 @@
 }
 
 static struct usb_driver synusb_driver = {
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,16)
-	.owner =	THIS_MODULE,
-#endif
 	.name =	"synaptics-usb",
 	.probe =	synusb_probe,
 	.disconnect =	synusb_disconnect,
